File: code/initialization.java
import java.util.Locale;

public class Initialization {
    static class InitTestClass {
        String someString1 = "Hello";
        int zahl = 5;
        {
            System.out.println("Die Zahl lautet: " + zahl);
            String someString2 = "World";
            System.out.println(someString1 + " " + someString2);
        }

        int zahlAdded2 = zahl + 2;
        {
            System.out.println("Die Zahl " + zahl + " + 2 lautet: " + zahlAdded2);
            System.out.println(someString1.toUpperCase(Locale.ROOT));
        }
    }

    public static void main(String[] args) {
        InitTestClass someObject = new InitTestClass();
        System.out.println(someObject);
    }
}
// Ausgabe:
// Die Zahl lautet: 5
// Hello World
// Die Zahl 5 + 2 lautet: 7
// HELLO

File: code/initialization.kt
// Initialization code can be placed in initializer blocks,
// which are prefixed with the init keyword.

// During an instance initialization,
// the initializer blocks are executed in the same order
// as they appear in the class body, interleaved with the property initializers:
class InitTestClass(zahl: Int) {
    // val and var that have been declared outside
    // of an init can be used by all following inits:
    val someString1 = "Hello"

    init {
        println("Die Zahl lautet: $zahl")
        // val and var, which were declared within an init,
        // can also only be used in this init:
        val someString2 = "World"
        println("$someString1 $someString2")
    }

    val zahlAdded2 = zahl + 2
    init {
        println("Die Zahl $zahl + 2 lautet: $zahlAdded2")
        println(someString1.toUpperCase())
    }

}

fun main () {
    // Executing the class InitTestClass:
    InitTestClass(5)
}
// Ausgabe:
// Die Zahl lautet: 5
// Hello World
// Die Zahl 5 + 2 lautet: 7
// HELLO